kafka/adminapi.go

```diff
@@ -990,6 +1070,55 @@ func (a *AdminClient) cToTopicResults(cTopicRes **C.rd_kafka_topic_result_t, cCn
 	return result, nil
 }
 
+// cToAuthorizedOperations converts a C AclOperation_t array to a Go
+// ACLOperation list.
+func (a *AdminClient) cToAuthorizedOperations(
+	cAuthorizedOperations *C.rd_kafka_AclOperation_t,
+	cAuthorizedOperationCnt C.size_t) []ACLOperation {
+	if cAuthorizedOperations == nil {
+		return nil
+	}
+
+	authorizedOperations := make([]ACLOperation, int(cAuthorizedOperationCnt))
+	for i := 0; i < int(cAuthorizedOperationCnt); i++ {
+		cAuthorizedOperation := C.AclOperation_by_idx(
+			cAuthorizedOperations, cAuthorizedOperationCnt, C.size_t(i))
+		authorizedOperations[i] = ACLOperation(cAuthorizedOperation)
+	}
+
+	return authorizedOperations
+}
+
+// cToNode converts a C Node_t* to a Go Node.
+func (a *AdminClient) cToNode(cNode *C.rd_kafka_Node_t) Node {
+	node := Node{ID: -1}
+	if cNode == nil {
+		return node
```

Comment:
Because we're returning a struct, not the pointer type `*Node`
As to why I don't return a pointer, rather than a struct, it's because I'm using this method in cToNodes which returns (and should return) []Node. 
I can make a separate one returning a pointer cToNodePtr or something

Reply:
I've changed it to create a new method, and removed the nil check from this one